packages/knip/src/plugins/react-router/index.ts

```diff
@@ -0,0 +1,65 @@
+import { existsSync } from 'node:fs';
+import type { IsPluginEnabled, Plugin, ResolveEntryPaths } from '../../types/config.js';
+import { toEntry } from '../../util/input.js';
+import { join } from '../../util/path.js';
+import { hasDependency, load } from '../../util/plugin.js';
+import type { PluginConfig, RouteConfigEntry } from './types.js';
+import vite from '../vite/index.js';
+
+// https://reactrouter.com/start/framework/routing
+
+const title = 'react-router';
+
+const enablers = ['@react-router/dev'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ['react-router.config.{js,ts}', ...vite.config];
```

Comment:
I added vite.config aswell because the `react-router.config` is optional.

Reply:
All `config` paths are loaded and if there's `resolveConfig` its executed with the default exported value of those config files. Also they're automatically added as `entry` files (so their imports are registered as dependencies like any other source file).

Since this plugin doesn't have `resolveConfig` I think `'react-router.config.{js,ts}'` should be moved to the `entry` array (guess I've missed this before), and looks like there's no value in adding `vite.config`.

Just saying, the `config` array items do not influence whether the plugin is enabled or not.

Reply:
Any chance you could move `config` to `entry` and remove `vite.config` - I'd guess that gives the same results, right?

Reply:
Moving `react-router.config.ts` to `entry` and removing vite doesn't seem to enable the plugin. When console logging inside `resolveEntryPaths` nothing happends. Not sure whats happening.

Reply:
Ah sorry I looked over the fact that `localConfig` is an argument to `resolveEntryPaths` and thus a `config` item is required. Took the liberty to clean up and merge :)